docs/architecture/modernize-desktop/deploying-modern-desktop-applications.md

```diff
@@ -0,0 +1,332 @@
+---
+title: Deploying Modern Desktop Applications
+description: Everything you need to know about deploying for modern desktop applications.
+ms.date: 09/16/2019
+---
+
+# Deploying Modern Desktop Applications
+
+## Introduction
+
+When you develop desktop applications, besides other development aspects, you
+also have to think about how your application is going to be packaged and
+deployed to the final users' machines. The problem with packaging, deployment,
+and installation is that it usually falls under the umbrella of the IT
+professionals who care about different things than developers.
+
+These days we are all familiar with the DevOps concept where developers and IT
+pros work closely to move applications to their production environments. But if
+you have been in the desktop battle for more than 10 years you might have witnessed a following tory. A team of developers works together hard to meet the
+project deadlines. Business people are nervous since they need the system working
+on numerous user's machines to run the company. On the "D-Day", the
+project manager checks with every developer that their code is working well,
+everything is fine, they can ship. Then the package team comes in generating the
+setup for the app, distribute it to every user machine and a set of
+test users run the application. Well, they try, because before showing any UI
+the application throws an exception that says "Method ~ of object ~ failed".
+Panic starts flowing through the air and a brief investigation points to a young
+and tired developer that has introduced a third-party control, that certainly "worked on the dev machine".
+
+Installing desktop applications have traditionally been of  a nightmare for
+two main reasons: a lack of culture of close collaboration between dev and IT
+teams and a lack of a solid packaging a deploying technology we can build upon.
+In fact, we have been leaving with the fact that sometimes you regret to install
+an app because it ends up having some undesired side effects and some already
+installed applications stop working. Moreover, you cannot just restore the
+system to its original state by uninstalling. We are so used to live this
+situation that we have coined terms like "DLL Hell" or "Winrot".
```

Comment:
```suggestion
Installing desktop applications have traditionally been a nightmare for
two main reasons: 

- Lack of close collaboration culture between dev and IT teams.
- Lack of a solid packaging and deploying technology we can build upon.

In fact, we've been living with the fact that sometimes you regret that you installed
an app because:

- It ends up having some undesired side effects on your machine.
- Some applications that were previously installed stop working. 

Additionally, you can't just restore the system to its original state by uninstalling the app. We're so used to live this situation that we've coined terms like "DLL Hell" or "Winrot".
```